### DISCUSSION TITLE
clarity on docs for setup of longhorn and disks

### DISCUSSION THREAD

**Discussion by tuesdaycodes** [longhorn/longhorn#3335] [Q&A]
do i run the install on all nodes or just the server node?
guessing setup of the disk in the ui is all i need and when i make the pv/pvc will use the block device. does the same block device need added to all nodes or do they iscsi/nfs from main server by default for persisted storage to deploys?

**Comment by derekbit** [comment]
@tuesdaycodes 
By default, the deployment yaml will help you to install to all nodes.
You can check the details of the design and usage on https://longhorn.io/docs/1.2.2/ .

**Reply by tuesdaycodes** [reply]
thx. oddly when i follow your docs or k3s sample config for longhorn my pvc isnt actually shared/persisted. if the app container moves all the data is wiped and the app needs setup again. just doing test cluster right now so no real data loss yet. same if i make it `ReadWriteMany` and see the share longhorn pod show up, more than 1 replica doesnt seem to use the same data.

**Reply by jenting** [reply]
> if the app container moves all the data is wiped and the app needs setup again

How do you perform the application container moves? By moving the Pod from Node 1 to Node 2?

**Reply by tuesdaycodes** [reply]
> > if the app container moves all the data is wiped and the app needs setup again
> 
> How do you perform the application container moves? By moving the Pod from Node 1 to Node 2?

by draining the node.

**Reply by joshimoo** [reply]
Are you using ephemeral volumes? Can you check the workload pvc that you are testing?
One other thing, if your deployment.yaml includes the PVC like this example here: https://longhorn.io/docs/1.2.2/references/examples/#pod-with-persistentvolumeclaim

When you do a `kubectl delete -f deployment.yaml` you will also delete the PVC / volume so when you then do a `kubectl create -f deployment.yaml` you  will create a brand new volume, can you check your deployment that you use for testing.

**Reply by joshimoo** [reply]
Also check that your pvc's are created with the `longhorn` storage class, it's possible that you are using the default storage class which is local path provisioner on k3s.
